packages/autocomplete-core/src/onInput.ts

```diff
@@ -90,9 +92,14 @@ export function onInput<TItem extends BaseItem>({
               state: store.getState(),
               ...setters,
             })
-          ).then((itemsOrDescription) =>
-            preResolve<TItem>(itemsOrDescription, source.sourceId)
-          );
+          ).then((itemsOrDescription) => {
+            invariant(
+              itemsOrDescription !== undefined,
+              'The `getItems` function should return or resolve to an array of items.'
+            );
```

Comment:
The existing invariant that we have [lives in `postResolve`](https://github.com/algolia/autocomplete/blob/3e785660d65d568e611542dec8de20eb87a001b0/packages/autocomplete-core/src/resolve.ts#L173-L178). For consistency, we can move that one to the `preResolve` function.